redux/ducks/collections.js

```diff
@@ -0,0 +1,66 @@
+import createReducer from 'type-to-reducer';
+
+import omit from 'lodash/omit';
+
+import api from 'constants/api';
+import { defaultReducer } from 'utils/redux';
+import { makeRequest } from 'utils/request';
+import { getLocalizedCollection, getLocalizedArticles, getArticlesBlocks } from 'utils/getters';
+
+const duck = 'collections';
+
+// constants
+const FETCH_COLLECTION = `${duck}/FETCH_COLLECTION`;
+
+// reducer
+const initialState = {
+  pending: false,
+  error: false,
+  collection: null,
+};
+
+export default createReducer(
+  {
+    [FETCH_COLLECTION]: defaultReducer((state, { payload }) => ({
+      ...state,
+      ...payload,
+    })),
+  },
+  initialState
+);
+
+// selectors
+const getState = state => state.collections;
+const getData = (state, lang) => {
+  const { collection } = getState(state);
+  const localizedArticles = getLocalizedArticles(collection.articles, lang);
+
+  const authors = Object.values(
+    localizedArticles.reduce((acc, { tagsByTopic: { authors: articleAuthors } }) => {
+      articleAuthors.forEach(authorTag => {
+        acc[authorTag.slug] = authorTag;
+      });
+      return acc;
+    }, {})
```

Comment:
alternative way:
```js
const authors = new Set();
localizedArticles.forEach(({ tagsByTopic }) => {
   tagsByTopic.authors.forEach(tag => {
       authors.add(tag);
   });
});
```

> [TL:DR Set is almost two times faster than Object](https://stackoverflow.com/questions/44243103/javascript-set-vs-array-vs-object-definition)

Reply:
I don't think it is a good idea to rely on uniqueness of objects (it's much more reliable to use slugs as keys). Also it does not seem to work: I've tried and it looks like object uniqueness is something different from deep value equality (probably addresses or smth like that).
![Screen Shot 2020-07-04 at 17 38 36](https://user-images.githubusercontent.com/7117196/86514811-3aea8b00-be1d-11ea-9d48-f3b2cbca82f8.png)